### ISSUE TITLE
Generated models not extending from base interface

### ISSUE THREAD

**Issue by manijak** [cyclosproject/ng-openapi-gen#177]
I am aware of this issue from before (marked as fixed): https://github.com/cyclosproject/ng-openapi-gen/issues/142
But I am having issues in generating models that extend from base interface. The base interface is imported in child interfaces, but they do not extend from it for some reason. 

**Schema:** 
```
"BaseObjDto": {
        "type": "object",
        "properties": {
          "type": {
            "type": "string",
            "nullable": true
          }
        },
        "additionalProperties": false
  },
 "ChildObjDto": {
        "type": "object",
        "allOf": [
          {
            "$ref": "#/components/schemas/BaseObjDto"
          }
        ],
        "properties": {
          "name": {
            "type": "string",
            "nullable": true
          }
        },
        "additionalProperties": false
},
"AnotherObjDto": {
        "type": "object",
        "allOf": [
          {
            "$ref": "#/components/schemas/BaseObjDto"
          }
        ],
        "properties": {
          "name": {
            "type": "string",
            "nullable": true
          }
        },
        "additionalProperties": false
},
```

**Generated models (base imported as reference but never extends):**

```
export interface BaseObjDto {
  type?: null | string;
}
```

```
import { BaseObjDto } from './base-obj-dto';
export interface ChildObjDto {
  name?: null | string;
}
```

```
import { BaseObjDto } from './base-obj-dto';
export interface AnotherObjDto {
  name?: null | string;
}
```

Any idea why this is happening?

**Comment by micaelboucard** [comment]
I'm having the same problem and got really confused on this topic as @manijak mentioned:
* there was a misalignment with [Swashbuckle peeps](https://github.com/domaindrivendev/Swashbuckle.AspNetCore/issues/1942)
* The OpenAPI specs is not clear on whether this definition is valid or not
* the issue was fixed in v0.16.0, but not working on v0.19.1 again

We would really appreciate your view on this @luisfpg ! 🙏🏽

**Comment by luisfpg** [comment]
Sorry @micaelboucard , but I don't get what is the problem.
What was fixed in 0.16.0 is that models didn't extend the other if they had `type: objectP` and `allOf` in the same level.
This currently works. I just added a test for the same models as https://github.com/cyclosproject/ng-openapi-gen/issues/177#issuecomment-952785173 and the result is:

```typescript
export interface Shape {
  name?: string;
}
```
and
```typescript
import { Shape } from './shape';
export type Circle = Shape & {
'radius'?: number;
};
```

So, `Circle` is actually a `Shape`.

Can you provide an example of exactly what isn't working as expected?

**Comment by micaelboucard** [comment]
> Sorry @micaelboucard , but I don't get what is the problem. What was fixed in 0.16.0 is that models didn't extend the other if they had `type: object` and `allOf` in the same level. This currently works. I just added a test for the same models as [#177 (comment)](https://github.com/cyclosproject/ng-openapi-gen/issues/177#issuecomment-952785173) and the result is:
> 
> ```ts
> export interface Shape {
>   name?: string;
> }
> ```
> 
> and
> 
> ```ts
> import { Shape } from './shape';
> export type Circle = Shape & {
> 'radius'?: number;
> };
> ```
> 
> So, `Circle` is actually a `Shape`.
> 
> Can you provide an example of exactly what isn't working as expected?

Hi @luisfpg !

The issue I was referring to, is the same one @manijak provided. For a simple example using:
- Data models using polymorphism
- Swashbuckle on C# service to generate OpenAPI specs
- ng-openapi-gen on Angular app to generate models and services

So Swashbuckle package defines it like this:

```
"components": {
"schemas": {
	"Fruit": {
		"type": "object",
		"properties": {
			"name": {
                      "type": "string",
                      "nullable": false
                  }
		},
              "required": [
                  "name"
              ],
		"additionalProperties": false
	},
	"Banana": {
		"type": "object",
		"allOf": [
			{
				"$ref": "#/components/schemas/Fruit"
			}
		],
              "properties": {
                  "length": {
                      "type": "string",
                      "nullable": false
                  }
              },
              "required": [
                  "length"
              ],
		"additionalProperties": false
	}
}
}
```

And `ng-openapi-gen` generates the 2 models as:

```
/* tslint:disable */
/* eslint-disable */
export interface Fruit {
  name: string;
}
```
and
```
/* tslint:disable */
/* eslint-disable */
import { Fruit } from './fruit';
export interface Banana {
  length: string;
}
```
missing the `... extends Fruit` typescript bit.

Unless that is not the way to define it?

**Comment by micaelboucard** [comment]
> Sorry @micaelboucard , but I don't get what is the problem. What was fixed in 0.16.0 is that models didn't extend the other if they had `type: object` and `allOf` in the same level. This currently works. I just added a test for the same models as [#177 (comment)](https://github.com/cyclosproject/ng-openapi-gen/issues/177#issuecomment-952785173) and the result is:
> 
> ```ts
> export interface Shape {
>   name?: string;
> }
> ```
> 
> and
> 
> ```ts
> import { Shape } from './shape';
> export type Circle = Shape & {
> 'radius'?: number;
> };
> ```
> 
> So, `Circle` is actually a `Shape`.
> 
> Can you provide an example of exactly what isn't working as expected?

Hey @luisfpg ! Thanks for your reply.
Sorry, I just saw that the issue was solved back again in recent releases `v0.21.0`.
Just tried with the latest one and everything works now 👏🏽 

Thank you for all what you do.
